### ISSUE TITLE
Explore `index` parameter of HDF5DataSet

### ISSUE THREAD

**Issue by gcroci2** [DeepRank/deeprank-core#159]
In `HDF5DataSet` class, `index` input parameter usage can be clarified. Docs say that it refers to the index of a molecule, and that defaults to None.

- For what is it used, and why is it needed?
- Add these info to the docs
- Add an example that makes clear a situation in which is convenient to use it

**Comment by cbaakman** [comment]
It seems to me that the index field of the HDF5DataSet class is used internally only. There's no reason why it should be an input argument.

**Comment by DaniBodor** [comment]
index needs to be an int, but the `create_index_molecules` method looks for index and tries to iterate over it, which is not possible for ints. This might be a moot point if the parameter is going to be removed anyway, but wanted to mention it because I guess this should have raised a red flag somewhere but it didn't. 
Unless I'm missing something here... @gcroci2

**Comment by DaniBodor** [comment]
I am a bit confused because `index` is used by different functions/methods as different things here, but here is what I think is happening:
- As far as I can tell `self.index` is only used to create the `mol_names` list, which is then used to fill the `index_complexes` list
- However, `index` should be a list, not an int for this. Indeed the readme example passes a range of to index, but the HDF5DataSet class explanation states that should be an int. So I guess just the documentation is wrong in this case. And naming it `index` if it is in fact a list of indices is confusing (at least to me).
- Within the HDF5DataSet class, the `get` method also uses a parameter called `index`, which should be an int. But this is a different `index` than `self.index`.

Bottom line, I think the index parameter is actually used and should stay but maybe renamed.

**Comment by DaniBodor** [comment]
To answer this question "Add an example that makes clear a situation in which is convenient to use it"

I guess it's if you only want to use part of the data, but not sure I understood this correctly and if there's a different way already there to select a subset of the data.
